### PR TITLE
Add tests to understand orient's concurrent modification behavior

### PR DESCRIPTION
**Pull request by cwilper**

### CODE REVIEW

File: components/nexus-orient/src/main/java/org/sonatype/nexus/orient/PersistentDatabaseInstanceRule.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    log.info("Database instance cleaned up");
+  }
+
+  class PersistentDatabaseManager
+      extends DatabaseManagerSupport {
+
+    private final File databasesDirectory;
+
+    PersistentDatabaseManager() {
+      File targetDir = new File(this.getClass().getProtectionDomain().getCodeSource().getLocation().getFile()).getParentFile();
```
Comment (kellyrob99) [inline components/nexus-orient/src/main/java/org/sonatype/nexus/orient/PersistentDatabaseInstanceRule.java]: probably can avoid all this and use TemporaryFolder, unless there's a compelling reason to keep the file system around after the test exits

**Comment by kellyrob99** [comment]
+1

**Comment by mcculls** [comment]
+1

**Comment by cwilper** [comment]
Oh btw, the reason i changed the junit rule to use a persistent db is that in some earlier manual testing, i noticed that orient behaved a bit differently when using a memory db (it doesn't bother to serialize/deserialize stored values, so using joda DateTime directly works when the db is memory-based, but not when it's persistent). Given that, I got nervous that concurrency tests might also behave differently in memory. So I figured the safest course of action would be to have these tests use a real persistent db which is closer to reality.

**Pull request merged**